integration_test.go

```diff
@@ -283,3 +286,89 @@ func TestDefaultValue(t *testing.T) {
 		return result.ID > 0
 	}, "ID should be greater than 0")
 }
+
+func TestDistinct(t *testing.T) {
+	skipIfShort(t)
+	t.Parallel()
+	dsn, cleanup, err := createTestDB(context.Background())
+	if err != nil {
+		log.Fatalf("could not init integration tests while creating database: %v", err)
+		os.Exit(1)
+	}
+	defer cleanup()
+	// Open db.
+	db, err := gorm.Open(New(Config{
+		DriverName: "spanner",
+		DSN:        dsn,
+	}), &gorm.Config{PrepareStmt: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := db.AutoMigrate(&testutil.User{}, &testutil.Account{}, &testutil.Pet{}, &testutil.Company{}, &testutil.Toy{}, &testutil.Language{},
+		&testutil.Coupon{}, &testutil.CouponProduct{}, &testutil.Order{}, &testutil.Parent{}, &testutil.Child{}); err != nil {
+		t.Fatalf("Failed to migrate models, got error: %v", err)
+	}
+
+	users := []testutil.User{
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct-2", testutil.Config{}),
+		*testutil.GetUser("distinct-3", testutil.Config{}),
+	}
+	users[0].Age = 20
+
+	if err := db.Create(&users).Error; err != nil {
+		t.Fatalf("errors happened when create users: %v", err)
+	}
+
+	var names []string
+	db.Table("users").Where("name like ?", "distinct%").Order("name").Pluck("name", &names)
+	testutil.AssertEqual(t, names, []string{"distinct", "distinct", "distinct", "distinct-2", "distinct-3"})
+
+	var names1 []string
```

Comment:
nit: `Get distinct name from the users table, using `Distinct` and `Pluck``. This is not very meaningful but having this comment would help someone later to understand what we are validating here.